File: go/easy/valid_palindrome.go
package main

import (
  "fmt"
  "regexp"
  "strings"
)

func isPalindrome(s string) bool {
  re := regexp.MustCompile(`[^a-zA-Z0-9]`)
  ns := re.ReplaceAllString(s, "")
  ns = strings.ToLower(ns)
  for i, j := 0, len(ns)-1; i <= len(ns)/2; i, j = i+1, j-1 {
    if ns[i] != ns[j] {
      return false
    }
  }
  return true
}

// func main() {
//   s := "A man, a plan, a canal: Panama"
//   fmt.Println(isPalindrome(s))
// }

File: python/easy/valid_palindrome.py
class Solution:
    # def isPalindrome(self, s: str) -> bool:
        # x = [c.lower() for c in s if c.isalnum()]
        # front = x[0:(len(x) // 2)]
        # back = x[-1:-(len(x) // 2 + 1):-1]
        # return (front == back)

    def isPalindrome(self, s: str) -> bool:
        x = [c.lower() for c in s if c.isalnum()]
        return x[0:(len(x) // 2)] == x[-1:-(len(x) // 2 + 1):-1]


test1 = "A man, a plan, a canal: Panama"
#true

test2 = "race a car"
#true

test3 = " "
#true

test4 = "bo oboo!B"

sol = Solution()
sol.isPalindrome(test4)